diffrax/_integrate.py

```diff
@@ -1084,6 +1084,22 @@ def _promote(yi):
     y0 = jtu.tree_map(_promote, y0)
     del timelikes
 
+    # Check if the solver is an instance of AbstractSolver and provide an informative
+    # error if it is not. Addresses https://github.com/patrick-kidger/diffrax/issues/705
+    if not isinstance(solver, AbstractSolver):
+        if issubclass(solver, AbstractSolver):
```

Comment:
This will error out if `solver` is not a class. We need `isinstance(solver, type) and issubclass(solver, AbstractSolver)`.